integration_tests/src/main/python/generate_expr_test.py

```diff
@@ -216,3 +216,36 @@ def test_generate_outer_fallback():
         lambda spark: spark.sql("SELECT array(struct(1, 'a'), struct(2, 'b')) as x")\
             .repartition(1).selectExpr("inline_outer(x)"),
         "GenerateExec")
+
+# gpu stack not guarantee to produce the same output order as Spark does
+@ignore_order(local=True) 
+def test_stack():
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark : spark.range(100).selectExpr('*', 'stack(3, id, 2L, 3L, 4L, 5L, 6L)'))
+
+# gpu stack not guarantee to produce the same output order as Spark does
+@ignore_order(local=True)
+def test_stack_mixed_types():
+    data_gen = StructGen([['child'+str(ind), sub_gen] for ind, sub_gen in 
+                          enumerate(all_basic_gens + decimal_gens)], nullable=False)
```

Comment:
nit: This test feels a little brittle because we have constants that are tied to the order of all_basic_gens and decimal_gens. Could you make a copy of them locally so that they don't change or generate the literal values from the datagen itself?

Reply:
Ok, will do in my next pr.